src/query/atomWithQuery.ts

```diff
@@ -98,16 +98,22 @@ export function atomWithQuery<
         const unsubscribe = observer.subscribe(listener)
         return unsubscribe
       }
-      return { dataAtom, options }
+      return { dataAtom, observer }
     },
     (get, set, action: AtomWithQueryAction) => {
       switch (action.type) {
         case 'refetch': {
-          const { dataAtom, options } = get(queryDataAtom)
+          const { dataAtom, observer } = get(queryDataAtom)
           set(dataAtom, new Promise<TData>(() => {})) // infinite pending
-          const queryClient = get(getQueryClientAtom)
-          queryClient.getQueryCache().find(options.queryKey)?.reset()
-          const p: Promise<void> = queryClient.refetchQueries(options.queryKey)
+          const p: Promise<void> = new Promise((resolve, reject) =>
+            observer
+              .refetch({ cancelRefetch: true })
+              .then(() => resolve())
+              .catch((error) => {
+                set(dataAtom, Promise.reject<TData>(error))
```

Comment:
Hm, are you sure we need this? You mean L69 doesn't work for refetch?

You don't need `new Promise`, something like this would do.
```js
return observer.refetch({ cancelRefetch: true }).catch((error) => {
  set(dataAtom, Promise.reject<TData>(error))
})
```
or
```js
return observer.refetch({ cancelRefetch: true }).then(() => {}).catch((error) => {
  set(dataAtom, Promise.reject<TData>(error))
})
```

Reply:
should be fixed, and I removed the error setting, was not needed